### DISCUSSION TITLE
E450 encrypted

### DISCUSSION THREAD

**Discussion by lastv8** [scs/smartmeter-datacollector#44] [General]
This is my 2nd trial to get smartmeter-datacollector running, but without success.
I have a E450. My local energy supplier activated the M-Bus interface and I got the GAK and GUEK keys.
I can see that data arrives. The led on the M-bus board is blinking regularly. I also activated MQTT.
Which key should I use? GAK or GUEK? I tried both.
Is there a way to check what data arrives? Any other hints?

**Comment by raymar9** [comment]
Hi lastv8
You need only the GUEK which is used to decrypt the data received from the meter. The GAK is only for authentication and not required / implemented in our software.
If you login to the raspberry pi you can inspect the log with
```
journalctl -u python3-smartmeter-datacollector [-f]
```
as described in our wiki: https://github.com/scs/smartmeter-datacollector/wiki/How-to-use#run-as-a-systemd-service .
If you set the log level to DEBUG beforehand you can even inspect the raw data that is received in the log.
